### ISSUE TITLE
Top bar position is reset after screen lock

### ISSUE THREAD

**Issue by iexos** [freddez/gnome-shell-simple-message#11]
### Steps to reproduce
- open settings and select "Center" in top panel location
- lock screen
- unlock
- message is moved to the right again, expected to be in the center
  - setting remains on center but is ignored until set again

### System
- Arch Linux
- Gnome 43.3
- Simple Message 13

**Comment by clover-yan** [comment]
@freddez @iexos Yes, the same on `Ubuntu 22.04`, `GNOME 42.9`, `Simple Message 14`.

**Comment by Anuneza** [comment]
I don’t maintain this extension, but I ran into the same behavior and was able
to work around it locally.

If you are using version 14 (which, as far as I can tell, I suppose it does not differ
significantly from v13 in this area), you can try editing extension.js and
changing the following lines.

Replace:

    Main.panel.addToStatusArea(
        'simpleMessage',
        this,
        this.message_position,
        this.message_alignment
    );

with:

    GLib.idle_add(GLib.PRIORITY_DEFAULT_IDLE, () => {
        this._moveMessage();
        return GLib.SOURCE_REMOVE;
    });

And replace:

    this.get_parent().remove_actor(this);

with:

    this.get_parent()?.remove_child(this);

This defers the insertion until after the panel is rebuilt (lock/unlock or
GNOME Shell reload), which seems to prevent the indicator from jumping to the
right.

System

Ubuntu 22.04.5 LTS (btw XDDD)
GNOME Shell 42.9
Simple Message 14

**Comment by freddez** [comment]
Hello Anuneza Thank you for this long waiting fix !
It is published in version  25
